{"@class": "de.fau.cs.inf2.cthree.data.Cluster", "id": "3d99070165e6ea0336c20575924eee81", "detectedBy": ["DIFF_HIERARCHICAL"], "members": [{"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "b75938344e1e0b7c53603656e496e7f5", "repository": "https://github.com/checkstyle/checkstyle.git", "fileName": "src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RedundantThrowsCheckTest.java", "commitBeforeChange": "3994e36e8d25c3e5bc8570d371fc73b1467cf80c", "commitAfterChange": "fcb47f13dc5876474d9cbf9c3f5e8949b0d67887", "methodNumberBeforeChange": 3, "methodNumberAfterChange": 3, "signatureBeforeChange": "      @Test     public void testAllowSubclasses() throws Exception", "signatureAfterChange": "      @Test     public void testAllowSubclasses() throws Exception", "diff": ["-            \"13:16: Redundant throws: 'RuntimeException' is unchecked exception.\",", "-            \"19:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "-            \"39:27: Redundant throws: 'NullPointerException' is unchecked exception.\",", "-            \"39:49: Redundant throws: 'RuntimeException' is unchecked exception.\",", "+            \"15:16: Redundant throws: 'RuntimeException' is unchecked exception.\",", "+            \"21:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "+            \"41:27: Redundant throws: 'NullPointerException' is unchecked exception.\",", "+            \"41:49: Redundant throws: 'RuntimeException' is unchecked exception.\","]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "cd18de10ef198885da3d5bf7279502d3", "repository": "https://github.com/checkstyle/checkstyle.git", "fileName": "src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RedundantThrowsCheckTest.java", "commitBeforeChange": "3994e36e8d25c3e5bc8570d371fc73b1467cf80c", "commitAfterChange": "fcb47f13dc5876474d9cbf9c3f5e8949b0d67887", "methodNumberBeforeChange": 1, "methodNumberAfterChange": 1, "signatureBeforeChange": "      @Test     public void testDefaults() throws Exception", "signatureAfterChange": "      @Test     public void testDefaults() throws Exception", "diff": ["-            \"7:37: Redundant throws: 'java.io.FileNotFoundException' is subclass of 'java.io.IOException'.\",", "-            \"13:16: Redundant throws: 'RuntimeException' is unchecked exception.\",", "-            \"19:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "-            \"39:27: Redundant throws: 'NullPointerException' is subclass of 'RuntimeException'.\",", "-            \"39:27: Redundant throws: 'NullPointerException' is unchecked exception.\",", "-            \"39:49: Redundant throws: 'RuntimeException' is unchecked exception.\",", "+            \"9:37: Redundant throws: 'java.io.FileNotFoundException' is subclass of 'java.io.IOException'.\",", "+            \"15:16: Redundant throws: 'RuntimeException' is unchecked exception.\",", "+            \"21:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "+            \"41:27: Redundant throws: 'NullPointerException' is subclass of 'RuntimeException'.\",", "+            \"41:27: Redundant throws: 'NullPointerException' is unchecked exception.\",", "+            \"41:49: Redundant throws: 'RuntimeException' is unchecked exception.\","]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "ffd4cc515f871e628e4f0c80b4d955e8", "repository": "https://github.com/checkstyle/checkstyle.git", "fileName": "src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RedundantThrowsCheckTest.java", "commitBeforeChange": "3994e36e8d25c3e5bc8570d371fc73b1467cf80c", "commitAfterChange": "fcb47f13dc5876474d9cbf9c3f5e8949b0d67887", "methodNumberBeforeChange": 2, "methodNumberAfterChange": 2, "signatureBeforeChange": "      @Test     public void testAllowUnchecked() throws Exception", "signatureAfterChange": "      @Test     public void testAllowUnchecked() throws Exception", "diff": ["-            \"7:37: Redundant throws: 'java.io.FileNotFoundException' is subclass of 'java.io.IOException'.\",", "-            \"19:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "+            \"9:37: Redundant throws: 'java.io.FileNotFoundException' is subclass of 'java.io.IOException'.\",", "+            \"21:29: Redundant throws: 'java.io.IOException' listed more then one time.\",", "-            \"39:27: Redundant throws: 'NullPointerException' is subclass of 'RuntimeException'.\",", "+            \"41:27: Redundant throws: 'NullPointerException' is subclass of 'RuntimeException'.\","]}]}